src/backend/renderer/element/surface.rs

```diff
@@ -305,21 +307,29 @@ where
 }
 
 /// A single surface render element
-pub struct WaylandSurfaceRenderElement<R> {
+pub struct WaylandSurfaceRenderElement<R: Renderer> {
     id: Id,
     location: Point<f64, Physical>,
     alpha: f32,
-    surface: wl_surface::WlSurface,
     renderer_type: PhantomData<R>,
     kind: Kind,
+
+    view: Option<SurfaceView>,
+    current_commit: CommitCounter,
```

Comment:
We do not have to store the `CommitCounter` separately from the `damage`, `DamageSnapshot::current_commit` can directly be used in `Element::current_commit`.